src/ufoLib2/objects/font.py

```diff
@@ -345,31 +377,35 @@ def guidelines(self):
         return self.info.guidelines
 
     @guidelines.setter
-    def guidelines(self, value):
+    def guidelines(self, value: List[Guideline]) -> None:
         for guideline in value:
             self.appendGuideline(guideline)
 
     @property
-    def path(self):
+    def path(self) -> Optional[PathLike]:
         """Return the path of the UFO, if it was set, or None."""
         return self._path
 
-    def addGlyph(self, glyph):
+    def addGlyph(self, glyph: Glyph) -> None:
         """Appends glyph object to the default layer unless its name is already
         taken.
 
         Note:
             Call the method on the layer directly if you want to overwrite entries
             with the same name or append copies of the glyph.
         """
+        if self.layers.defaultLayer is None:
+            raise Error("No default layer present, cannot append Glyph object.")
         self.layers.defaultLayer.addGlyph(glyph)
 
-    def newGlyph(self, name):
+    def newGlyph(self, name: str) -> Glyph:
         """Creates and returns new :class:`.Glyph` object in default layer with
         name."""
+        if self.layers.defaultLayer is None:
```

Comment:
all this checking that defaultLayer is not None is getting repetitive. When is defaultlayer None?

Reply:
I don't know, but the type is Optional. Should I look into dropping Optional and requiring a default layer? I think the spec says there must be one layer at least, but it does not have to be named "public.default" 🤔